src/main/java/seedu/address/logic/commands/MassRejectCommand.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.JobCode;
+import seedu.address.model.person.JobCodePredicate;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Tag;
+
+/**
+ * MassRejectCommand changes the tags of persons in bulk based on
+ * the specified job code and tag criteria.
+ */
+public class MassRejectCommand extends Command {
+
+    public static final String COMMAND_WORD = "massReject";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Rejects persons with specified job code and/or tag.\n"
+            + "Parameters:\n"
+            + "JobCode + Tag: " + COMMAND_WORD + " j/SWE2024 t/bp\n"
+            + "Tag only: " + COMMAND_WORD + " t/bp\n"
+            + "JobCode only (excluding accepted): " + COMMAND_WORD + " j/SWE2024\n";
+
+    public static final String MESSAGE_SUCCESS = "Mass rejection complete. %d persons updated.";
+
+    private final Predicate<Person> predicate;
+
+    public MassRejectCommand(Predicate<Person> predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // Step 1: Use the predicate to filter the relevant persons
+        List<Person> personsToReject = model.getFilteredPersonList().stream()
+                .filter(person -> shouldReject(person))
+                .collect(Collectors.toList());
+
```

Comment:
Good job using predicates to filter persons! Have you considered using the `model.updateFilteredPersonList(predicate)` method directly, which could simplify filtering? This may also align with other parts of the app where predicates are used.